src/stream-tcp-reassemble.c

```diff
@@ -1039,14 +1039,8 @@ static StreamingBufferBlock *GetBlock(const StreamingBuffer *sb, const uint64_t
     if (blk == NULL)
         return NULL;
 
-    for ( ; blk != NULL; blk = SBB_RB_NEXT(blk)) {
-        if (blk->offset >= offset)
-            return blk;
-        else if ((blk->offset + blk->len) > offset) {
-            return blk;
-        }
-    }
-    return NULL;
+    StreamingBufferBlock key = { .offset = offset, .len = 0 };
+    return SBB_RB_FIND_INCLUSIVE((struct SBB *)&sb->sbb_tree, &key);
```

Comment:
Seems to be working well on all cases. 🎉

Reply:
🎉🎉 
Will this PR be merged?

Reply:
yes if it passes our internal QA. This will however likely be looked at after the upcoming Suricata release. Thank you for your work! :)